snapshot-labs/sx-monorepo
# perf: disable notifications loading for guest user

### Summary



Working towards https://github.com/snapshot-labs/workflow/issues/142

This PR disable the notifications loading/polling for guest user.

By combining everything inside `watch`, we can later disable the whole notifications logic by simply adding a guard inside this watcher

### How to test

1. Apply diff

```diff --git a/apps/ui/src/stores/notifications.ts b/apps/ui/src/stores/notifications.ts
index 7aab4118..6e65ca6e 100644
--- a/apps/ui/src/stores/notifications.ts
+++ b/apps/ui/src/stores/notifications.ts
@@ -13,7 +13,7 @@ type Notification = {
 };

 const NOTIFICATION_TIME_WINDOW = 60 * 60 * 24 * 14; // 2 weeks
-const REFRESH_INTERVAL = 60 * 15; // 15 minutes
+const REFRESH_INTERVAL = 1; // 15 minutes
 let refreshNotificationInterval: number;

 export const useNotificationsStore = defineStore('notifications', () => {
@@ -43,6 +43,7 @@ export const useNotificationsStore = defineStore('notifications', () => {
   }

   async function loadNotifications() {
+    console.log('loading notifications');
     const now = Math.floor(Date.now() / 1000);
     const pivotTs = now - NOTIFICATION_TIME_WINDOW;
```

2. `yarn dev`
3. Go to http://localhost:8080/
4. As a guest user, it should not print the console message
5. Log in
6. You should start seeing the console.log every 1s
7. Log out
8. The console message should stop printing
